### PR TITLE
xtensa/esp32s3: Add support for Tickless kernel using Systimer

### PR DESCRIPTION
**Pull request by gustavonihei**
## Summary
This PR intends to add support for Tickless operation of the NuttX kernel using the Systimer component.

## Impact
Extended OS support for `esp32s3-devkit` platform.

## Testing
Tested using `esp32s3-devkit:tickless`.
Boot into NSH and successful execution of `ostest`.

### CODE REVIEW

File: arch/xtensa/src/esp32s3/esp32s3_tickless.c [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+ *
+ * Input Parameters:
+ *   None.
+ *
+ * Returned Value:
+ *   Total system ticks.
+ *
+ ****************************************************************************/
+
+static inline uint64_t up_tmr_getcounter(void)
```
Comment (pkarashchenko) [inline arch/xtensa/src/esp32s3/esp32s3_tickless.c:122]: why do you use `up_` notation for local functions?

Comment (gustavonihei) [reply arch/xtensa/src/esp32s3/esp32s3_tickless.c:122]: Interesting question. This driver is a port from another ESP32 chip, the `up_` prefix didn't caught my attention before.
It seems pointless for these internal functions to be prefixed with `up_`. I will rename them.
Thanks.

Comment (pkarashchenko) [reply arch/xtensa/src/esp32s3/esp32s3_tickless.c:122]: Thank you

Comment (gustavonihei) [reply arch/xtensa/src/esp32s3/esp32s3_tickless.c]: Done, renamed the private `up_` prefixed functions to `tickless_`.

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**